controllers/config_helper.go

```diff
@@ -0,0 +1,192 @@
+var configDefaultResource = []byte(`
+apiVersion: config.gatekeeper.sh/v1alpha1
+kind: Config
+metadata:
+  name: config
+spec:
+  match:
+  - excludedNamespaces: ["kube-*", "multicluster-engine", "hypershift", "hive", "rhacs-operator", "open-cluster-*"] 
+    processes: ["audit", "webhook", "sync"]
+`)
+
+func createConfigFromYaml(ctx context.Context, c client.Client, namespace string) error {
+	config := &v1alpha1.Config{}
+
+	err := yaml.Unmarshal(configDefaultResource, config)
+	if err != nil {
+		return err
+	}
+
+	config.ObjectMeta = metav1.ObjectMeta{
+		Name:      "config",
+		Namespace: namespace,
+	}
+
+	err = c.Create(ctx, config)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkConfigHasExemptNs(ctx context.Context,
+	c client.Client, existingConfig v1alpha1.Config, log logr.Logger,
+) error {
+	configDefault := &v1alpha1.Config{}
+
+	err := yaml.Unmarshal(configDefaultResource, configDefault)
+	if err != nil {
+		return err
+	}
+
+	// Check only ExcludedNamespaces not processes
+	// Users can edit processes
+
+	for _, match := range existingConfig.Spec.Match {
+		if reflect.DeepEqual(match,
+			configDefault.Spec.Match[0]) {
+			log.Info("Default excludedNamespaces exist already")
+
+			return nil
+		}
+	}
+
+	existingConfig.Spec.Match = append(existingConfig.Spec.Match, configDefault.Spec.Match[0])
+
+	err = c.Update(ctx, &existingConfig, &client.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	log.Info("Update config because Default excludedNamespaces were missed")
+
+	return nil
+}
+
+func (r *GatekeeperReconciler) handleConfigController(mainCtx context.Context) error {
+	isCRDReady, err := checkConfigCrdAvailable(mainCtx, r.DynamicClient)
+	if err != nil {
+		return err
+	}
+
+	if !isCRDReady {
+		return errCpsCrdNotReady
+	}
+
+	if r.isConfigCtrlRunning {
+		return nil
+	}
+
+	configMgr, err := ctrl.NewManager(r.KubeConfig, ctrl.Options{
+		Scheme: r.Scheme,
+		Metrics: server.Options{
+			BindAddress: "0",
+		},
+		LeaderElection:   r.EnableLeaderElection,
+		LeaderElectionID: "config.gatekeeper.operator",
+	})
+	if err != nil {
+		setupLog.Error(err, "Failed to setup NewManager for Config controller")
+
+		return err
+	}
+
+	if err := (&ConfigReconciler{
+		Client:    configMgr.GetClient(),
+		Log:       ctrl.Log.WithName("Config"),
+		Namespace: r.Namespace,
+	}).SetupWithManager(configMgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "Config")
+
+		return err
+	}
+
+	r.isConfigCtrlRunning = true
+
+	// Use another go routine for the Config controller
+	go func() {
+		err := configMgr.Start(mainCtx)
+		if err != nil {
+			setupLog.Error(err, "A problem running Config manager. Triggering a reconcile to restart it.")
+		}
+
+		r.isConfigCtrlRunning = false
+	}()
+
+	return nil
+}
+
+// Check Config Crd status is "True" and type is "NamesAccepted"
+func checkConfigCrdAvailable(mainCtx context.Context, dynamicClient *dynamic.DynamicClient) (bool, error) {
```

Comment:
This seems to duplicate the check for the ConstraintPodStatus CRD. Could you please consolidate this to a helper function?

Reply:
It is looking at Config CRD. should I consolidate?

Reply:
Yeah, but it's the same checks. I suggest a single function that accepts the CRD name as a parameter.